ESMF/GOCART2G_GridComp/GOCART2G_GridCompMod.F90

```diff
@@ -1062,22 +1055,30 @@ subroutine createInstances_ (self, GC, rc)
 !   Active instances must be created first! This ordering is necessary for
 !   filing the AERO states that are passed to radiation.
 !   This is achieved by arranging the names of the active instances first.
+    dso_libname = adjust_dso_name('libDU2G_GridComp')
```

Comment:
@weiyuan-jiang You probably don't need to do this. MAPL_AddChildFromDSO should adjust it for you automatically. So you could just pass in, say, `libDU2G_GridComp.so`.  But maybe this is better? Hmm.

That said, note that the CI is failing because I'm not sure we expose `adjust_dso_name` publicly in MAPL. Might be a question for @tclune if it should be or not.

Reply:
Ahh, you are right. It is adjusted inside the program. I will clean this up

Reply:
This exercise is in part to expose issues at various stages of build/run.    Relocatability is the big concern that a poorly understand.